packages/repository/src/utils/utils.ts

```diff
@@ -0,0 +1,60 @@
+import type { DataRecordValue, LyricRepositoryConfig, SongRepositoryConfig } from '@overture-stack/maestro-common';
+
+/**
+ * Finds the repository by its repository code
+ * @param repositories
+ * @param repoCode
+ * @returns
+ */
+export const getRepoInformation = (
+	repositories: (SongRepositoryConfig | LyricRepositoryConfig)[],
+	repoCode: string,
+): SongRepositoryConfig | LyricRepositoryConfig | undefined => {
+	return repositories.find((repo) => repo.code === repoCode);
+};
+/**
+ * Determines if passed configuration is appropiate for Lyric
+ * @param object
+ * @returns
+ */
+export const isLyricConfiguration = (
+	object: LyricRepositoryConfig | SongRepositoryConfig,
+): object is LyricRepositoryConfig => {
+	return object.type === 'LYRIC';
+};
+
+/**
+ * Determines if passed configuration is appropiate for Song
+ * @param object
+ * @returns
+ */
+export const isSongConfiguration = (
+	object: LyricRepositoryConfig | SongRepositoryConfig,
+): object is SongRepositoryConfig => {
+	return object.type === 'SONG';
+};
```

Comment:
Since TS 5.5 you shouldn't need to declare the type assertion for this predicate, if this type narrowing is valid then TS will infer it on its own. (ie. remove `object is songRepositoryConfig`.

Since this is just using differentiated union on `object.type` you could probably replace these functions with a `switch(repository.case)` in the code that uses these.